pkg/beacon/relay/groupselection/groupselection.go

```diff
@@ -27,21 +32,6 @@ type Result struct {
 // CandidateToNewGroup attempts to generate and submit tickets for the staker to
 // join a new group.
 //
-// There are two phases of ticket submission:
-// - initial ticket submission,
-// - reactive ticket submission.
-//
-// During the initial ticket submission, only tickets with a value below the
-// natural threshold are submitted to the chain. Those tickets have the highest
-// chance of being selected to the group and this way we minimize staker's
-// gas expenditure.
-//
-// During the reactive ticket submission, all other staker's tickets are
-// submitted. Reactive ticket submission is skipped if during the initial
-// ticket submission there was enough tickets submitted to a chain to form
-// a group. Those tickets could be submitted by any stakers participating in
-// a new group selection.
-//
 // The function never submits more tickets than the group size.
 func CandidateToNewGroup(
```

Comment:
It's worth adding documentation explaining how ticket submission works. We can borrow [the one](https://github.com/keep-network/keep-core/pull/1510/files#r402287010) from the chain.